image.go

```diff
@@ -63,6 +66,26 @@ func (i *image) Config(ctx context.Context) (ocispec.Descriptor, error) {
 	return i.i.Config(ctx, provider, platforms.Default())
 }
 
+func (i *image) IsUnpacked(ctx context.Context, snapshotterName string) (bool, error) {
+	sn := i.client.SnapshotService(snapshotterName)
```

Comment:
should we move this logic to `images.Check`?

If so, may have to create a return type for it

Reply:
/cc @stevvooe

Reply:
Check is a really busy method already with all those return results.

Reply:
could have it return a type, like I originally suggested. But it kinda feel that `Check` should give me that too.

Reply:
I can update the Check return results to be a type - would we want that in this same PR?

Reply:
Can be a separate PR

Reply:
This needs to be added as a separate helper function for the image type in the `images` package. Once we review that, we can add the sugar in the containerd client.

I'm still not thrilled with this never ending interface model -- it doesn't scale as functionality changes. See the go-grpc package for a reference as to where this leads. We need to writing helpers that are a function of the type, rather than adding methods to the type itself, unless it operates on a core or hidden abstraction.

Reply:
> This needs to be added as a separate helper function for the image type in the images package. Once we review that, we can add the sugar in the containerd client.

The current unpack logic is not put into the `images` package and I think it is better that way so that we don't introduce the `snapshot` package interfaces to the `images` package. This would also introduce the naming patterns used by the clients for referencing snapshots, which is more opinionated than what should be included in the `images` package. If we want to keep the implementations for these helpers out of the root, we could probably move `Unpack` and this new function to `rootfs`.

> I'm still not thrilled with this never ending interface model

I agree the `Image` interface does not seem to going in the right direction. Is there a need for this interface as all, why don't we just export `image` and use `*Image` from the client as well. Then we could be more flexible in adding methods or helpers for that type.

Reply:
Just chatted with @dmcgowan about this. Proposed solution to have the `Unpack` function take in a `SnapshotterService` and `containerd.Image` - that way it is no longer a member function. Thoughts?